### PR TITLE
Added ability to shutdown nodejs servers gracefully 

### PR DESCRIPTION
**Pull request by paltanmoy**
1) Gracefully shutting down APIServer
2) Gracefully shutting down ServiceBroker

### CODE REVIEW

File: api/index.js [hunk 1]
```diff
@@ -1,7 +1,31 @@
...
+
+var gracefulShutdown = function(signal) {
+  logger.info("Received " + signal + " signal, shutting down gracefully...");
+  apiServer.shutdown(function() {
+    logger.info('Everything is cleanly shutdown');
+  })
+}
+
+//listen for SIGTERM signal e.g. kill
+process.on ('SIGTERM', function(){
```
Comment (fraenkel) [inline api/index.js]: We shouldn't be doing a graceful shutdown on SIGTERM. That should be the escape valve for kill immediate.

File: servicebroker/lib/index.js [hunk 2]
```diff
@@ -1,7 +1,30 @@
...
+
+var gracefulShutdown = function(signal) {
+  logger.info("Received " + signal + " signal, shutting down gracefully...");
+  server.shutdown(function() {
+    logger.info('Everything is cleanly shutdown');
+  })
+}
+
+//listen for SIGTERM signal e.g. kill
+process.on ('SIGTERM', function(){
```
Comment (fraenkel) [inline servicebroker/lib/index.js]: See above

File: api/index.js (hunk 1 above)
Comment (paltanmoy) [inline api/index.js]: SIGTERM removed

**Comment by fraenkel** [comment]
Normally there should be a test added especially for the keep-alive, but we can go without it for now.

**Pull request merged**